### PR TITLE
Setup Traefik docker-compose

### PR DESCRIPTION
**Pull request by alexandroqc**
## Creating docker-compose file for Traefik reverse proxy.
  - Let's Encrypt TLS certificates enabled 
  - HTTP to HTTPS redirection

### CODE REVIEW

File: traefik/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - --entryPoints.http.address=:80
+      - --entryPoints.https.address=:443
+      ## Logging
+      - --log=true
+      - --log.level=INFO
+      - --accesslog=true
+      ## API
+      - --api=true
+      - --api.dashboard=true
+      - --api.insecure=true
```
Comment (s2-t2) [inline traefik/docker-compose.yaml]: comment this

File: traefik/docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      - --providers.docker.endpoint=unix:///var/run/docker.sock
+      - --providers.docker.network=reverse-proxy
+      - --providers.docker.exposedByDefault=false
+      ## Metrics
+      - --metrics.prometheus=true
+      - --metrics.prometheus.buckets=0.1,0.3,1.2,5.0
+      # Certificates
+      - --certificatesResolvers.le.acme.httpchallenge=true
+      - --certificatesResolvers.le.acme.httpchallenge.entrypoint=http
+      - --certificatesResolvers.le.acme.email=pedro.parqueador@galacticcoders.xyz
```
Comment (s2-t2) [inline traefik/docker-compose.yaml]: maybe make this an environment variable?

File: traefik/docker-compose.yaml (hunk 1 above)
Comment (alexandroqc) [reply traefik/docker-compose.yaml]: Done

File: traefik/docker-compose.yaml (hunk 2 above)
Comment (alexandroqc) [reply traefik/docker-compose.yaml]: Sounds good.
Done

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+traefik/.env*
```
Comment (s2-t2) [inline .gitignore:1]: :+1:

File: traefik/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      - --providers.docker.network=reverse-proxy
+      - --providers.docker.exposedByDefault=false
+      ## Metrics
+      - --metrics.prometheus=true
+      - --metrics.prometheus.buckets=0.1,0.3,1.2,5.0
+      # Certificates
+      - --certificatesResolvers.le.acme.httpchallenge=true
+      - --certificatesResolvers.le.acme.httpchallenge.entrypoint=http
+      - --certificatesResolvers.le.acme.email={LE_EMAIL}
+      - --certificatesResolvers.le.acme.storage=/letsencrypt/acme.json
```
Comment (sujata5538) [inline traefik/docker-compose.yaml:30]: :thinking:

**Review by s2-t2 — APPROVED** [review]
LGTM

**Pull request merged**